src/common/mfu_flist.c

```diff
@@ -684,6 +684,54 @@ uint64_t mfu_flist_file_get_perm(mfu_flist bflist, uint64_t idx) {
     return mode & (S_IRWXU | S_IRWXG | S_IRWXO);
 }
 
+#if DCOPY_USE_XATTRS
+void *mfu_flist_file_get_acl(mfu_flist bflist, uint64_t idx, ssize_t *acl_size, char *type)
+{
+    flist_t* flist = (flist_t*) bflist;
+    const char* filename = mfu_flist_file_get_name(flist, idx);
+    size_t val_bufsize = 1024;
+    void* val = (void*) MFU_MALLOC(val_bufsize);
+    ssize_t val_size;
+    int got_val = 0;
+
+    *acl_size = 0;
+
+    while(!got_val) {
```

Comment:
You can get rid of `got_val` if you make this a `while (retries < maxRetries)` loop and place a break statements in the `else` block that sets `got_val = 1`. You can then bound the loop with the retries to make sure the loop terminates.